Melanzana.MachO/MachLinkEditData.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Melanzana.Streams;
+
+namespace Melanzana.MachO
+{
+    /// <summary>
+    /// Linker data representation.
+    /// 
+    /// This holds data for relocations, symbol tables, string tables and various
+    /// other linking related information. It's either part of the __LINKEDIT
+    /// segment for fully linked files, or appended at the end of the object file
+    /// for unlinked Mach-O files.
+    /// </summary>
+    public class MachLinkEditData
```

Comment:
TODO: I likely need to change this a bit to have access to `Is64Bit` and `IsLittleEndian`. That will require a bit of shuffling of where `MachObjectFile` is passed to constructors now.

Reply:
Still not sure about this one. I will mark some constructors as internal and add `MachObjectFile` parameter to others to future proof this.